trc/restart_trcrd.F90

```diff
@@ -86,9 +90,9 @@ subroutine restart_trcrd(rstfnm_ocn)
 !
   call restart_ocntrcrd(rstfnm_ocntrc)
 
-#ifdef HAMOCC
-  call hamocc_init(1,rstfnm_hamocc)
-#endif
+  if (use_HAMOCC) then
+     call hamocc_init(1,rstfnm_hamocc)
+  end if
```

Comment:
```suggestion
  if (use_HAMOCC) then
     call hamocc_init(1,rstfnm_hamocc)
  end if
```

The CI intel compiler build fails on the call to hamocc_init. When meson builds without the 'ecosys' option it does not compile code in the `hamocc` directory, so then `hamocc_init` is not available.

Reply:
I tried to update the ci.yml file to include tests with 'ecosys' enabled, but the tests are still failing. Not sure what's going on, but the meson build system doesn't seem to recognize `ecosys=true`.

Reply:
@TomasTorsvik - thanks for looking into this. I agree that it makes sense at this point to keep the #ifdef HAMOCC on.
Are we okay with merging this PR - or do we want to resolve the meson build system problem first?

Reply:
We can merge this now. I can add the meson build afterwards, I didn't enable it properly with ecosys==true.